src/core/lib/lattice/hal/default/dcrtpoly.cpp

```diff
@@ -2694,25 +2697,28 @@ void DCRTPolyImpl<VecType>::FastBaseConvqToBskMontgomery(
     }
 
     // mod mtilde = 2^16
-    std::vector<uint16_t> result_mtilde(n);
+    uint64_t mtilde      = (uint64_t)1 << 16;
```

Comment:
mtilde and mtilde_half should be declared "const", so the compiler may be able to optimize the code a bit better  (http://www.gotw.ca/gotw/081.htm)

Reply:
Fixed as suggested.